### DISCUSSION TITLE
Extend Config Search Path to Structured Configs

### DISCUSSION THREAD

**Discussion by mbignotti** [facebookresearch/hydra#2589] [Q&A]
Hi!
I'm developing an Hydra app that will provide a configuration structure defined at the app level. Then, users, who will install the app with pip, will be allowed to extend the provided config groups with their own objects. For example, the app will define the `model` config group, and the user will add the config for their own models to that group.
In order to do so, I have implemented a config `SearchPathPlugin` that closely follows what has been done in issue #2001.
However, I would like to let the user decide whether to use yaml or structured configs. I've managed to make it work with yaml configs, but I'm struggling to make it work with structured ones. I've tried to place the extra config in a folder with `__init__.py`, place the actual config inside `__init__.py` and use `pkg://...` as a search path. Then, when I run `python my_app.py --help` I can see the extra config in the available config group options, but when I try to use it by selecting the option with the CLI, it crashes.
Any help would be highly appreciated!
Thanks a lot!

**Comment by omry** [comment]
Hydra does not have a built-in mechanism to discover Structured Configs.
The ConfigStore is a singleton and is utilized regardless of the config searchpath.
It's up to you to ensure that `ConfigStore.store()` method is called for all needed configs prior to the app composing configs utilizing them.

**Reply by mbignotti** [reply]
Ok. So I guess that the only way to achieve what I'm trying to do is exposing a path to a function, that can be overridden, where the user specifies all his structured configs. Then that function will always be called in the main.py file before @hydra.main. 
Or maybe it's possible to create something fancier. In any case, I cannot take advantage of the config search path mechanism.

Thanks for the clarification!

**Reply by omry** [reply]
Yes.
I did consider introducing something like this when I first worked on Structured Configs but decided that it was already very complicated and  that the ConfigStore would provide a sufficiently flexible solution that is also simple.
